cmd/kubelet/app/server.go

```diff
@@ -126,7 +126,7 @@ func UnsecuredKubeletDeps(s *options.KubeletServer) (*kubelet.KubeletDeps, error
 
 	var dockerClient dockertools.DockerInterface
 	if s.ContainerRuntime == "docker" {
-		dockerClient = dockertools.CreateDockerClientOrDie(s.DockerEndpoint, s.RuntimeRequestTimeout.Duration)
+		dockerClient = dockertools.ConnectToDockerOrDie(s.DockerEndpoint, s.RuntimeRequestTimeout.Duration)
```

Comment:
We don't want, but we have to now.
We must set correct docker client api version so that docker can talk with daemon with correct version of remote api. Because we support multiple docker versions, we can only get it from docker daemon and update our client.